xbmc/filesystem/FileDirectoryFactory.cpp

```diff
@@ -64,20 +64,22 @@ IFileDirectory* CFileDirectoryFactory::Create(const CURL& url, CFileItem* pItem,
 
   std::string strExtension=URIUtils::GetExtension(url);
   StringUtils::ToLower(strExtension);
-  VECADDONS codecs;
-  CAddonMgr::GetInstance().GetAddons(codecs, ADDON_AUDIODECODER);
-  for (size_t i=0;i<codecs.size();++i)
+  if (!strExtension.empty())
```

Comment:
Regardless, there should be a registry of sort here, that is responsible of keeping track of all these extensions and protocols in a sane manner. You should _not_ rely on CAddonMgr for this. Additionally, this very similar looking code is smeared all over the place. With InputStream and vfs addons it's copy-pasted into even more places. <- that we need to fix.